lib/active_model/serializer/caching.rb

```diff
@@ -208,46 +199,44 @@ def object_cache_key(serializer, adapter_instance)
         end
       end
 
-      def cached_attributes(fields, cached_attributes, adapter_instance)
+      ### INSTANCE METHODS
+      def fetch_attributes(fields, cached_attributes, adapter_instance)
         if self.class.cache_enabled?
```

Comment:
Yeah. Ultimately, I'd like to move a lot of this into a class that lives outside of the serializer.